### PR TITLE
fix warnings

### PR DESCRIPTION
**Pull request by ShishirRijal**
Fixed the warnings throughout the projects. 

Issue: #667

### CODE REVIEW

File: lib/application/configure_new_cbj_comp/configure_new_cbj_comp_bloc.dart [hunk 1]
```diff
@@ -234,7 +234,8 @@ class ConfigureNewCbjCompBloc
   ) async {
     // TODO: Not suer if it will open the WiFi when hotspot is open, delete if not
 
-    await WiFiForIoTPlugin.loadWifiList();
+    // await WiFiForIoTPlugin.loadWifiList();
```
Comment (guyluz11) [inline lib/application/configure_new_cbj_comp/configure_new_cbj_comp_bloc.dart:237]: Did you test that?

File: lib/application/light_toggle/light_toggle_bloc.dart [hunk 2]
```diff
@@ -58,6 +58,7 @@ class LightToggleBloc extends Bloc<LightToggleEvent, LightToggleState> {
     ChangeState event,
     Emitter<LightToggleState> emit,
   ) async {
+    // ignore: unused_local_variable
     Either<DevicesFailure, Unit> actionResult;
```
Comment (guyluz11) [inline lib/application/light_toggle/light_toggle_bloc.dart:62]: Var can be removed.

File: lib/application/plus_button_settings/plus_button_settings_bloc.dart [hunk 3]
```diff
@@ -37,7 +37,8 @@ class PlusButtonSettingsBloc
     Emitter<PlusButtonSettingsState> emit,
   ) async {
     emit(PlusButtonSettingsState.loadInProgress());
-    await _hubConnectionRepository.closeConnection();
-    context.router.replace(const ConnectToHubRoute());
+    await _hubConnectionRepository
+        .closeConnection()
+        .whenComplete(() => context.router.replace(const ConnectToHubRoute()));
```
Comment (guyluz11) [inline lib/application/plus_button_settings/plus_button_settings_bloc.dart:42]: It's the same logic, there is no need to change it.
Also the await make sure that the lines are more readable and less indented forward

File: lib/application/printers/printers_actor/printers_actor_bloc.dart [hunk 4]
```diff
@@ -75,21 +74,21 @@ class PrintersActorBloc extends Bloc<PrintersActorEvent, PrintersActorState> {
...
-      );
-    } else {
-      logger.e('Printer does not have lastKnownIp');
+    // final String printerIp = event.printer.deviceLastKnownIp.getOrCrash();
+    // if (printerIp != null) {
+    //   launchUrl(
+    //     Uri.parse('http://$printerIp'),
+    //     mode: LaunchMode.externalApplication,
+    //   );
+    // } else {
```
Comment (guyluz11) [inline lib/application/printers/printers_actor/printers_actor_bloc.dart:83]: You can't just comment code if it has warnings.

You have distorted a feature that easily like that

**Review by guyluz11 — COMMENTED** [review]
1. You are changing to many things in one pr, please split it to small pr with changes for the same type of warnings.
Logger deprecation in one pr, functions deprecation in another pr, indent in one pr, remove unused vars in one pr, and so on ...
2. Please don't change code without understanding the change, you are breaking features and probably didn't test thoroughly the consequences of each change

**Comment by ShishirRijal** [comment]
@guyluz11  I'm sorry for this. I thought we need to fix mentioned issues of one issue in one PR. New to this so. 
I'll try splitting them in small parts.  Thank you for the response.

File: lib/application/printers/printers_actor/printers_actor_bloc.dart (hunk 4 above)
Comment (ShishirRijal) [reply lib/application/printers/printers_actor/printers_actor_bloc.dart:83]: the variable was not-nullable. So I didn't see a point of null check.

File: lib/application/plus_button_settings/plus_button_settings_bloc.dart (hunk 3 above)
Comment (ShishirRijal) [reply lib/application/plus_button_settings/plus_button_settings_bloc.dart:42]: I was trying to fix the warning that says to not use the build context across async gaps.

**Pull request closed without merging**